src/docfx/build/Builder.cs

```diff
@@ -15,24 +15,31 @@ internal class Builder
         private readonly string _workingDirectory;
         private readonly CommandLineOptions _options;
         private readonly Watch<DocsetBuilder[]> _docsets;
+        private readonly Package _docsetPackage;
 
-        public Builder(ErrorBuilder errors, string workingDirectory, CommandLineOptions options)
+        public Builder(ErrorBuilder errors, string workingDirectory, CommandLineOptions options, Package docsetPackage)
         {
             _workingDirectory = workingDirectory;
             _options = options;
             _errors = errors;
+            _docsetPackage = docsetPackage;
             _docsets = Watcher.Create(LoadDocsets);
         }
 
-        public static bool Run(string workingDirectory, CommandLineOptions options)
+        public static bool Run(string workingDirectory, CommandLineOptions options, Package? docsetPackage = null)
```

Comment:
by default, it will be a null, and a localPackage wil be created to use, but for language server and our future spec test, a memory package will be passed into this function.
I can remove it for now, if you think it is confused, and add back when it is used